### PR TITLE
Only necessary files are published to npm

### PR DESCRIPTION
**Pull request by tomscholz**
Switched from .npmignore to files array inside package.json
- https://github.com/npm/npm/wiki/Files-and-Ignores
- https://docs.npmjs.com/files/package.json

### CODE REVIEW

**Comment by mourner** [comment]
Thanks for the PR! Is there any advantage to specifying included files in `package.json` as opposed to excluded files in `.npmignore`? It seems to me that it would be much easier to accidentally screw up a release with the first approach by forgetting to update `package.json` after a change in file paths.

**Comment by DannyDelott** [comment]
The general reason is that dotfiles are easy to miss when downloading/sharing source code or copying directories around.

There is also an argument to suggest that dotfiles are better for machine-specific config (eg: `.git`, `.vimrc`) and essential project config (eg: `package.json`, `yarn.lock`) belongs in files with more visibility.

File: package.json [hunk 1]
```diff
@@ -128,5 +128,12 @@
...
     "prepublish": "in-publish && run-s build-dev build-min || not-in-publish",
     "postinstall": "cd src/style-spec && yarn || npm install"
-  }
+  },
+  "files": [
+    "src/",
+    "dist/",
+    "plugins/",
+    "LICENSE.txt",
+    "ARCHITECTURE.md"
```
Comment (mollymerp) [inline package.json]: I don't think we'd want to include `ARCHITECTURE.md` in the npm package. We should include `README.md` and `CHANGELOG.md` though. 

Also the `plugins/` folder isn't needed (it actually shouldn't even be in `master` anymore)

Comment (tomscholz) [inline package.json]: Readme and changelog are automatically added see https://docs.npmjs.com/files/package.json#files

Comment (tomscholz) [inline package.json]: I also remove the `LICENSE.txt`.
In the npm documentation it states 
> Certain files are always included, regardless of settings:
[...]
LICENSE / LICENCE
README, CHANGES, LICENSE & NOTICE can have any case and extension.

**Comment by mollymerp** [comment]
You've still got `/plugins` in there 😸

**Comment by tomscholz** [comment]
Done! 💪🏻

**Comment by mollymerp** [comment]
@anandthakker do you know if we need to include the `flow_interfaces` directory in the files published to the npm registry? From what I can tell flow is used primarily for testing and we `unflowify` before we build the `dist` files, but I'm not sure if we need the flow files for people using mapbox-gl with another bundler 💭

**Comment by anandthakker** [comment]
> do you know if we need to include the flow_interfaces directory in the files published to the npm registry? From what I can tell flow is used primarily for testing and we unflowify before we build the dist files, but I'm not sure if we need the flow files for people using mapbox-gl with another bundler

Yeah, IMO we should keep these in so that people who want to bundle our source directly have a fighting chance.

**Comment by LitoMore** [comment]
We should not publish the `src` to npm.

**Comment by anandthakker** [comment]
@LitoMore we intentionally publish `src` to NPM so that developers who prefer to use flow-annotated sources are able to do so.

**Pull request merged**